sled-agent/health-monitor/src/handle.rs

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use illumos_utils::svcs::{SvcNotRunning, Svcs};
+use slog::Logger;
+use slog::error;
+use tokio::sync::watch;
+
+#[derive(Debug, Clone)]
+pub struct HealthMonitorHandle {
+    // TODO-K: Do I actually need the logger here?
+    pub log: Logger,
+    pub smf_services_in_maintenance_tx: watch::Sender<Vec<SvcNotRunning>>,
+}
+
+impl HealthMonitorHandle {
+    pub fn new(log: &Logger) -> Self {
+        // TODO-K: Does this new() make sense? Or just spawn_health
+        let (smf_services_in_maintenance_tx, _rx) =
+            watch::channel(vec![SvcNotRunning::new()]);
+        Self { log: log.clone(), smf_services_in_maintenance_tx }
+    }
+
+    // TODO-K: better return type with more health check fields?
+    pub fn to_inventory(&self) -> Vec<SvcNotRunning> {
+        // TODO-K: Add some logging?
+        self.smf_services_in_maintenance_tx.borrow().clone()
+    }
+}
+
+// TODO-K: Put this in another file?
+pub async fn poll_smf_services_in_maintenance(
+    log: Logger,
+    smf_services_in_maintenance_tx: watch::Sender<Vec<SvcNotRunning>>,
+) {
+    // We poll every minute to verify the health of all services. This interval
+    // is arbitrary.
+    const SVCS_POLL_INTERVAL: tokio::time::Duration =
+        tokio::time::Duration::from_secs(60);
+
+    loop {
+        match Svcs::enabled_not_running(&log).await {
+            Err(e) => error!(log, "failed to check SMF services' health"; &e),
```

Comment:
Can we report these errors in inventory too? If we had a higher-level health check struct, something like this?

```rust
struct SledAgentHealthCheckInventory {
    smf_services_in_maintenance: Result<Vec<SvcNotRunning>, SmfError>,
}
```

(Although serializing `Result`s in OpenAPI is kind of a pain, so we have some helpers lurking around for it.)

Reply:
Sure! That makes sense. That said, I think you reviewed an older commit because I do have a `HealthMonitorInventory` struct here 😄  -> https://github.com/oxidecomputer/omicron/pull/9434/files

Reply:
Made this change. I think like how it looks

```console
$ curl -H "api-version: 11.0.0"  http://[::1]:46440/inventory | jq
<...>
  "health_monitor": {
    "smf_services_in_maintenance": {
      "ok": {
        "services": [
          {
            "fmri": "svc:/site/fake-service2:default",
            "zone": "global"
          },
          {
            "fmri": "svc:/site/fake-service:default",
            "zone": "global"
          }
        ],
        "time_of_status": "2025-12-10T08:35:10.604488858Z"
      }
    }
  }
}
```

Didn't finish with everything else though. I'll address your other comments tomorrow :)